eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClientBuilder.java

```diff
@@ -95,15 +139,50 @@ public EventHubClientBuilder retry(RetryPolicy retryPolicy) {
     }
 
     /**
-     * Creates a new {@link EventHubClient} based on the configuration set in this builder.
+     * Sets the configuration store that is used during construction of the service client.
      *
-     * @return A new {@link EventHubClient} instance.
+     * The default configuration store is a clone of the {@link ConfigurationManager#getConfiguration() global
+     * configuration store}, use {@link Configuration#NONE} to bypass using configuration settings during construction.
+     *
+     * @param configuration The configuration store used to
+     * @return The updated EventHubClientBuilder object.
      */
-    public EventHubClient build() {
-        return new EventHubClient();
+    public EventHubClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
     }
 
-    //TODO (conniey): Remove placeholder when the client is updated.
-    public static class ProxyConfiguration {
+    private ProxyConfiguration constructDefaultProxyConfiguration(Configuration configuration) {
+        ProxyAuthenticationType authentication = ProxyAuthenticationType.NONE;
+        if (this.proxyConfiguration != null) {
+            authentication = this.proxyConfiguration.authentication();
+        }
+
+        String proxyAddress = configuration.get(BaseConfigurations.HTTP_PROXY);
+        String host;
```

Comment:
These variables (host, port, and proxy) can live inside the `if (proxyAddress != null) {` block. They aren't used outside of its scope.